core/common/src/main/java/alluxio/uri/URI.java

```diff
@@ -150,9 +149,11 @@ public static URI create(URI parent, URI child) {
       try {
         // To be compatible with URI, must use the last component of the scheme.
         parentUri = new java.net.URI(getSchemeComponents(parent.getScheme()).getSecond(),
-            parent.getAuthority(), parentPath, parent.getQuery(), null);
+            parent.getAuthority().toString().equals("") ? null : parent.getAuthority().toString(),
```

Comment:
do we need to include the authority in the `java.net.URI`s here? It would be much more efficient to push one of the authorities directly into the created URI instead of having to use toString, then parse to java.net.URI, then parse to alluxio.URI

Reply:
Changed to 
```
      Authority authority = child.getAuthority() instanceof NoAuthority
          ? parent.getAuthority() : child.getAuthority();
      try {
        // To be compatible with URI, must use the last component of the scheme.
        parentUri = new java.net.URI(getSchemeComponents(parent.getScheme()).getSecond(),
            null, parentPath, parent.getQuery(), null);
        childUri = new java.net.URI(getSchemeComponents(child.getScheme()).getSecond(),
            null, child.getPath(), child.getQuery(), null);
      } catch (URISyntaxException e) {
        throw new IllegalArgumentException(e);
      }
      java.net.URI resolved = parentUri.resolve(childUri);
```
directly get Authority from parent or child.
Avoid inefficient transformation, thanks